python/cuml/experimental/fil/fil.pyx

```diff
@@ -1228,6 +1263,106 @@ class ForestInference(UniversalBase, CMajorInputTagMixin):
             any power of 2, but little benefit is expected above a chunk size
             of 512.
         """
+        chunk_size = self._get_chunk_size(chunk_size)
         return self.forest.predict(
             X, predict_type="per_tree", preds=preds, chunk_size=chunk_size
         )
+
+    def optimize(
+        self,
+        *,
+        data=None,
+        batch_size=1024,
+        iterations=10,
+        predict_method='predict',
+        max_chunk_size=None,
+        seed=0
+    ):
+        """
+        Find the optimal layout and chunk size for this model
+
+        The optimal value for layout and chunk size depends on the model,
+        batch size, and available hardware. In order to get the most
+        realistic performance distribution, example data can be provided. If
+        it is not, random data will be generated based on the indicated batch
+        size. After finding the optimal layout, the model will be reloaded if
+        necessary. The optimal chunk size will be used to set the default chunk
+        size used if none is passed to the predict call.
+
+        Parameters
+        ----------
+        data
+            Example data of shape iterations x batch size x features or None.
+            If None, random data will be generated instead.
+        batch_size : int
+            If example data is not provided, random data with this many rows
+            per batch will be used.
+        iterations : int
+            The number of times to repeat inference in order to mitigate
+            outliers in performance data.
```

Comment:
Maybe document the default here? Alternatively, we could use a dynamic number of iterations like [autorange()](https://docs.python.org/3/library/timeit.html#timeit.Timer.autorange).

Reply:
I quite like the thought of using an autorange-like approach. That gives the caller some certainty about how long the optimization is likely to take. We can even let them control the target elapsed time in that case.